File: week 6/sentimental-readability/readability.py
import string
import math

# Get text
text = input("Text: ")

# Count spaces, letters, words, and sentences
spaces = text.count(" ")
letters = len([char for char in text if char.isalpha()])
sentence = text.count(".") + text.count("?") + text.count("!")
words = spaces + 1

# Calculate L, S, and index
L = (letters / words) * 100
S = (sentence / words) * 100
subindex = 0.0588 * L - 0.296 * S - 15.8
index = round(subindex)

# Print grade
if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int spaces = 0;
    int letters = 0;
    int words = 0;
    int sentence = 0;

    string text = get_string("Text: ");

    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (isspace(text[i]))
        {
            spaces++;
        }
        else if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentence++;
        }
    }
    words = spaces + 1;

    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentence / (float) words) * 100;
    float subindex = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(subindex);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
